packages/common/src/ui/components/modals/ConfirmationModal/ConfirmationModalProvider.tsx

```diff
@@ -13,14 +13,20 @@ interface ConfirmationModalState {
   message: string;
   title: string;
   iconType?: 'alert' | 'info';
-  onConfirm: (() => void) | (() => Promise<void>);
+  onConfirm?: (() => void) | (() => Promise<void>);
```

Comment:
it troubles me, having the `onConfirm` as optional, but I don't see a good alternative.
could require passing in `onConfirm` when triggering I suppose - wondering if that would be clearer overall